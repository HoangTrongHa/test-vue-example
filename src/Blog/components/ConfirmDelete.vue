<template>
  <v-row justify="center">
    <v-dialog :value="dialog" max-width="400" @input="updateDialog">
      <v-card>
        <v-card-title class="text-h5"> Chuẩn Chưa </v-card-title>

        <v-card-actions>
          <v-spacer></v-spacer>

          <v-btn color="green darken-1" text @click="updateDialog(false)">
            Chưa
          </v-btn>

          <v-btn color="green darken-1" text @click="confirm"> Chuẩn </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</template>
<script lang="ts">
import { Component, Vue, Prop, Emit } from "vue-property-decorator";
// import Employee from "../../types/Employee";
// import EmployeeDataService from "../../business/B_employee"

@Component
export default class Alert extends Vue {
  @Prop() readonly dialog?: boolean;

  @Emit("update-dialog")
  updateDialog(e: boolean) {
    return e;
  }

  @Emit("confirm")
  confirm() {
    this.updateDialog(false);
  }
}
</script>

