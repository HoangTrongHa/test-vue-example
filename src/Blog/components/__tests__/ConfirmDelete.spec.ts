import { shallowMount } from '@vue/test-utils'
import ConfirmDelete from '@/Blog/components/ConfirmDelete.vue'
import flushPromises from 'flush-promises';

function factory() {
  return shallowMount(ConfirmDelete, {
    stubs: ['v-row', 'v-dialog', 'v-card-title', 'v-card-actions', 'v-spacer', 'v-btn']
  });
}
describe('ConfirmDelete.vue', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('should match snapshot', async () => {
    const wrapper = factory();
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  })
})
