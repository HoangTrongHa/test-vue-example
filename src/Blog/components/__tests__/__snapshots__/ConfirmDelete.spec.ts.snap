// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDelete.vue should match snapshot 1`] = `
<v-row-stub
  justify="center"
>
  <v-dialog-stub
    max-width="400"
  >
    <v-card>
      <v-card-title-stub
        class="text-h5"
      >
         Chuẩn Chưa 
      </v-card-title-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          color="green darken-1"
          text=""
        >
          
          Chưa
        
        </v-btn-stub>
         
        <v-btn-stub
          color="green darken-1"
          text=""
        >
           Chuẩn 
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card>
  </v-dialog-stub>
</v-row-stub>
`;
